### PR TITLE
Build issues

### PR DESCRIPTION
**Pull request by barbelity**

### CODE REVIEW

File: artifactory/services/buildinfo.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+func (bis *BuildInfoService) GetArtifactoryDetails() auth.ArtifactoryDetails {
+	return bis.ArtDetails
+}
+
+func (bis *BuildInfoService) SetArtifactoryDetails(rt auth.ArtifactoryDetails) {
+	bis.ArtDetails = rt
+}
+
+func (bis *BuildInfoService) GetJfrogHttpClient() (*rthttpclient.ArtifactoryHttpClient, error) {
```
Comment (eyalbe4) [inline artifactory/services/buildinfo.go]: The error returned value seems redundant.

File: artifactory/services/buildinfo.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	"encoding/json"
+	"fmt"
+	"github.com/jfrog/jfrog-client-go/artifactory/auth"
+	"github.com/jfrog/jfrog-client-go/artifactory/buildinfo"
+	rthttpclient "github.com/jfrog/jfrog-client-go/artifactory/httpclient"
+	"github.com/jfrog/jfrog-client-go/artifactory/services/utils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+)
+
+type BuildInfoService struct {
```
Comment (eyalbe4) [inline artifactory/services/buildinfo.go]: I think we should merged the existing *PublishBuildInfoService* into this service.

File: utils/io/fileutils/files.go [hunk 3]
```diff
@@ -16,7 +17,11 @@ import (
 	"strings"
 )
 
-const SYMLINK_FILE_CONTENT = ""
+const (
+	SYMLINK_FILE_CONTENT                  = ""
+	File                 TraverseItemType = "file"
+	Folder               TraverseItemType = "dir"
```
Comment (eyalbe4) [inline utils/io/fileutils/files.go]: I suggest using *Dir* instead of *Folder*.

File: utils/io/fileutils/files.go [hunk 4]
```diff
@@ -400,3 +405,62 @@ func CopyDir(fromPath, toPath string, includeDirs bool) error {
 	}
 	return err
 }
+
+// Returns the path to the directory in which itemToFind is located.
+// Traversing through directories from current work-dir to root.
+// traverseType determines whether looking for a file or dir.
+func GetFileOrDirPath(itemToFInd string, traverseItemType TraverseItemType) (string, error) {
```
Comment (eyalbe4) [inline utils/io/fileutils/files.go]: I suggest renaming this function to *FindUpstream*.

File: utils/io/fileutils/files.go [hunk 5]
```diff
@@ -400,3 +405,62 @@ func CopyDir(fromPath, toPath string, includeDirs bool) error {
...
+		// If we already visited this directory, it means that there's a loop and we can stop.
+		if visitedPaths[wd] {
+			return "", errorutils.CheckError(errors.New(fmt.Sprintf("Could not find %s.", itemToFInd)))
+		}
+	}
+
+	return "", errorutils.CheckError(errors.New(fmt.Sprintf("Could not find %s.", itemToFInd)))
+}
+
+type TraverseItemType string
```
Comment (eyalbe4) [inline utils/io/fileutils/files.go]: I suggest renaming *TraverseItemType* to *ItemType* (in comments as well).

File: artifactory/services/buildinfo.go (hunk 1 above)
Comment (barbelity) [reply artifactory/services/buildinfo.go]: The BuildInfoService struct implements the CommonConf interface so this is mandatory.

File: utils/io/fileutils/files.go (hunk 4 above)
Comment (barbelity) [reply utils/io/fileutils/files.go]: Renamed.

File: utils/io/fileutils/files.go (hunk 3 above)
Comment (barbelity) [reply utils/io/fileutils/files.go]: Changed.

File: utils/io/fileutils/files.go (hunk 5 above)
Comment (barbelity) [reply utils/io/fileutils/files.go]: Done.

File: artifactory/services/buildinfo.go (hunk 2 above)
Comment (barbelity) [reply artifactory/services/buildinfo.go]: Merge the two services into BuildInfoService.

**Pull request merged**